### PR TITLE
Update validate_holdout_datasets docstring and docs

### LINKED ISSUE #3270: Improve documentation for validate_holdout_datasets
After #3208 the helper `validate_holdout_datasets` is only used in `test_gen_utils.py`. Let's delete it unless there's a justifiable reason to keep it. If we do keep it, let's improve the docstring since its currently vague: "Validate the holdout datasets match our expectations."

### PR DESCRIPTION
**Pull request by ParthivNaresh**
Fixes #3270 

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Review by freddyaboulton — APPROVED** [review]
Thanks @ParthivNaresh ! Looks good to me!

File: docs/source/user_guide/timeseries.ipynb [hunk 1]
```diff
@@ -293,10 +293,15 @@
...
     "\n",
     "* ``predict_in_sample`` is used when the target value is known on the dates we are predicting on. This is true in cross validation. This method has an expected ``y`` parameter so that we can compute features using previous target values for all of the observations on the holdout set.\n",
-    "* ``predict`` is used when the target value is not known, e.g. the test dataset. The y parameter is not expected as only the target is observed in the training set. The test dataset must be separated by ``gap`` days from the training dataset. For the moment, the test set size be less than or equal to ``forecast_horizon``.\n",
-    "\n",
-    "Here is an example of these two methods in action:\n",
+    "* ``predict`` is used when the target value is not known, e.g. the test dataset. The y parameter is not expected as only the target is observed in the training set. The test dataset must be separated by ``gap`` units from the training dataset. For the moment, the test set size be less than or equal to ``forecast_horizon``.\n",
```
Comment (jeremyliweishih) [inline docs/source/user_guide/timeseries.ipynb:296]: "For the moment, the test set size be less than or equal to ``forecast_horizon``" ----> "For the moment, the test set size **must** be less than or equal to ``forecast_horizon``"?

File: docs/source/user_guide/timeseries.ipynb [hunk 2]
```diff
@@ -309,6 +314,34 @@
     "pl.predict_in_sample(X_test, y_test, objective=None, X_train=X_train, y_train=y_train)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Validating the holdout data\n",
+    "\n",
+    "Before we predict on our holdout data, validating that it meets the requirements we summarized in the second point above in **Predicting on sunseen data** is important. We can call on `validate_holdout_datasets` in order to verify the two requirements:\n",
```
Comment (jeremyliweishih) [inline docs/source/user_guide/timeseries.ipynb:323]: nit: "Before we predict on our holdout data, validating that it meets the requirements we summarized in the second point above" ----> ""Before we predict on our holdout data, it is import to validate that it meets the requirements we summarized in the second above in ..."

File: docs/source/user_guide/timeseries.ipynb [hunk 3]
```diff
@@ -309,6 +314,34 @@
...
+    "\n",
+    "1. The holdout data is separated by `gap` units from the training dataset. This is determined by the `time_index` column, not the index e.g. if your datetime frequency for the column \"Date\" is 2 days with a `gap` of 3, then the holdout data must start 2 days x 3 = 6 days after the training data.\n",
+    "2. The holdout data must be less than or equal to the `forecast_horizon`."
```
Comment (jeremyliweishih) [inline docs/source/user_guide/timeseries.ipynb:326]: "The holdout data  **size** must.."

**Review by jeremyliweishih — APPROVED** [review]
Looks great! Some nits

**Pull request merged**